libbpf-rs/src/object.rs

```diff
@@ -334,6 +353,58 @@ impl Object {
         self.map(name).unwrap().unwrap()
     }
 
+
+    pub fn get_maps(&mut self) -> Result<&HashMap<String, Map>> {
```

Comment:
I think we should hide `HashMap` implementation detail from the user in case we ever want to change the implementation without breaking API. What do you think about returning a struct that implements `std::iter::Iterator`? Like what https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.iter does.

```suggestion
    pub fn get_maps(&mut self) -> Result<Iter<(String, Map)>> {
```

Reply:
Hm, for my personal project (not published yet) I get a lot of use from iterating and getting both the map name and the `Map` itself. I suppose I could just do `Map.name()` instead of unpacking the `HashMap` tuple.

Are you suggesting we just return `self.maps.iter()`? It might have to be `iter_mut()` if you want to do something with the maps, but same point I believe.

Reply:
Yeah, the iterator element type could be `(&str, &mut Map)`. And probably  not exactly `self.maps.iter()` b/c that would return a HashMap type. Reason would be in the event of us wanting to move from HashMap to something else, we wouldn't want to create a temporary HashMap just so we don't break API.

I'm suggesting we create our own type, `[Open]MapIter` and `[Open]ProgIter` (can use a macro to generate these) and return those instead.

Reply:
I'm afraid my greeness in Rust will show a bit here because I don't know how to use a macro to generate these types. I'm not sure even what to research about it. I'll have a look around when I get some time though. It would be nifty to use a macro instead of rolling it by hand.

Reply:
I will probably have time to get to this tomorrow.

Reply:
Ugh sorry about the long turnaround time. I'm in the middle of real life stuff (moving). I **will** get to this though.